src/lib/Instance.ts

```diff
@@ -0,0 +1,24 @@
+export default class Instance {
+  private _onChanged: ((...args: any[]) => void)[];
+  onChanged: {
+    connect: (callback: (...args: any[]) => void) => () => void;
+    fire: (...args: any[]) => void;
+  };
+
+  constructor() {
+    this._onChanged = [];
+    this.onChanged = {
+      connect: (callback: (...args: any[]) => void) => {
+        const length = this._onChanged.length;
+        this._onChanged.push(callback);
+
+        return () => {
+          this._onChanged.splice(length, 1);
+        };
+      },
+      fire: (...args: any[]) => {
+        this._onChanged.forEach((callback) => callback(args));
+      },
+    };
```

Comment:
Should be moved to a separate event class.
Event class should use generics to denote types passed to `.fire(...args)`, which are in turn passed to the function passed to `.connect()`